registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java

```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
 		assertFalse(messageDto.getIsValid());
 		assertFalse(messageDto.getInternalError());
 	}
+//    One match of RID/AID found at bio dedupe stage.
+//    UIN of the duplicate match is same as the update packet UIN
+    @Test
+    public void oneMatchFoundWithSameUin() throws ApisResourceAccessException, IOException,
+            PacketDecryptionFailureException, io.mosip.kernel.core.exception.IOException, JsonProcessingException, PacketManagerException {
+        registrationStatusDto.setRegistrationId("27847657360002520190320095011");
+        registrationStatusDto.setRegistrationType("UPDATE");
+        registrationStatusDto.setStatusCode("PROCESSING");
+        Mockito.when(registrationStatusService.getRegistrationStatus(any(),any(),any(), any())).thenReturn(registrationStatusDto);
+        Mockito.when(abisHandlerUtil.getPacketStatus(any())).thenReturn(AbisConstant.POST_ABIS_IDENTIFICATION);
+        Set<String> matchedRidList = new HashSet<>();
+        matchedRidList.add("27847657360002520190320095011");
+        matchedRidList.add("27847657360002520190320095012");
+        Mockito.when(abisHandlerUtil.getUniqueRegIds(any(), any(), anyInt(), any(), any())).thenReturn(matchedRidList);
+        Mockito.when(utility.getUIn(anyString(), anyString(), any())).thenReturn("9403107396");
+        Mockito.when(idRepoService.getUinByRid(anyString(),any())).thenReturn("9403107396");
+        ArgumentCaptor<InternalRegistrationStatusDto> argumentCaptor = ArgumentCaptor.forClass(InternalRegistrationStatusDto.class);
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        Mockito.verify(registrationStatusService).updateRegistrationStatus(argumentCaptor.capture(),anyString(),anyString());
+        Assert.assertEquals(argumentCaptor.getValue().getStatusCode(), RegistrationStatusCode.PROCESSING.name());
+        assertTrue(messageDto.getIsValid());
+        assertFalse(messageDto.getInternalError());
+    }
+
+//    One match of RID/AID found at bio dedupe,
+//    but UIN of the duplicate match is different from update packet UIN when AutoReject is True
+    @Test
+    public void oneMatchFoundWithDiffrentUinWithMatchAutoRejecAsTrue() throws ApisResourceAccessException, IOException,
+            PacketDecryptionFailureException, io.mosip.kernel.core.exception.IOException, JsonProcessingException, PacketManagerException {
+        registrationStatusDto.setRegistrationId("27847657360002520190320095011");
+        registrationStatusDto.setRegistrationType("UPDATE");
```

Comment:
Add the below for currently processing RID.
registrationStatusDto.setStatusCode("PROCESSING");
Check other occurrences as well.

Reply:
done